snafu/benchmarks/dnsperf/dnsperf.py

```diff
@@ -0,0 +1,300 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""Wrapper for running the dnsperf benchmark.
+See https://dns-oarc.net/tools/dnsperf for more information."""
+import os
+import dataclasses
+from datetime import datetime
+from typing import Any, Dict, Iterable, Optional
+from pathlib import Path
+
+import numpy as np
+import dateutil.parser
+from ttp import ttp
+
+from snafu.benchmarks import Benchmark, BenchmarkResult
+from snafu.config import Config, ConfigArgument
+from snafu.process import sample_process, ProcessSample, check_file
+
+
+@dataclasses.dataclass
+class RawDnsperfSample:
+    fqdn: str
+    rtt_mu_s: int
+    qtype: str
+    rcode: int
+
+    def __str__(self) -> str:
+        return (
+            f"Name: {self.fqdn}, "
+            f"Qtype: {self.qtype}, "
+            f"Rcode: {self.rcode}, "
+            f"Rtt microseconds: {self.rtt_mu_s}"
+        )
+
+    def __repr__(self) -> str:
```

Comment:
Was the goal of adding the ``__repr__`` method here to ensure that the dataclass had a repr method, or was it to override the one that dataclasses autogenerate?

Reply:
The goal was to override the autogenerated repr method because it did not pretty print the `data` object..